packages/asset-swapper/src/swap_quoter.ts

```diff
@@ -161,10 +165,13 @@ export class SwapQuoter {
         this.orderbook = orderbook;
         this.expiryBufferMs = expiryBufferMs;
         this.permittedOrderFeeTypes = permittedOrderFeeTypes;
+        this.rfqtTakerApiKeyWhitelist = options.rfqtTakerApiKeyWhitelist || [];
+        this.rfqtMakerEndpoints = options.rfqtMakerEndpoints || [];
         this._contractAddresses = options.contractAddresses || getContractAddressesForChainOrThrow(chainId);
         this._devUtilsContract = new DevUtilsContract(this._contractAddresses.devUtils, provider);
         this._protocolFeeUtils = new ProtocolFeeUtils(constants.PROTOCOL_FEE_UTILS_POLLING_INTERVAL_IN_MS);
         this._orderStateUtils = new OrderStateUtils(this._devUtilsContract);
+        this._quoteRequestor = options.quoteRequestor || new QuoteRequestor(this.rfqtMakerEndpoints);
```

Comment:
For testing purposes it may be a lot easier to take in an instantiated `QuoteRequestor` as a param, instead of constructing it in the class from `this.rfqtMakerEndpoints`.  

I don't have a good sense of the other changes that would be needed to support that though.  Do you think it may be worthwhile or not?

Reply:
I'm a little confused.  We _are_ enabling the client to pass in an instantiated `QuoteRequestor`.  But, it's within the options constructor param rather than having its own constructor param.

Having said that, while I wrote this a mind for injecting that dependency from the test scenario, but I didn't actually write a test to use that mechanism, so I may have been off base here.

Reply:
Maybe we should just scrap the whole notion.  I only wrote it that way because I thought we'd mock out the QuoteRequestor instance, to be able to test everything outside of it.  But if we're mocking out the Axios layer, then our tests can be exercising the QuoteRequestor as well, at which point there's no need to inject this dependency into the SwapQuoter.